db/document.go

```diff
@@ -527,14 +536,32 @@ func (s *SyncData) GetSyncCas() uint64 {
 	return base.HexCasToUint64(s.Cas)
 }
 
+func HasUserXattrChanged(rawUserXattr []byte, rawUserXattrHash string) bool {
```

Comment:
The parameter names here don't do a great job at identifying that they correspond to two different user xattrs (current and previous), and I don't think 'raw' is really needed for either of them.  Maybe userXattr and prevUserXattrHash?